src/thor/route_action.cc

```diff
@@ -436,8 +437,10 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
         vias.swap(flipped);
 
         // Form output information based on path edges
-        if (api.trip().routes_size() == 0 || api.options().alternates() > 0)
+        if (api.trip().routes_size() == 0 || api.options().alternates() > 0) {
           route = api.mutable_trip()->mutable_routes()->Add();
+          route->mutable_legs()->Reserve(correlated.size());
```

Comment:
Could do a little better here by filtering out locations that are `via`.

Reply:
that is true but what we are optimizing for here is avoiding reallocations. since the number of legs will always be less than or equal to correlated.size() this will accomplish that goal even if it does over estimate how much we need